### ISSUE TITLE
Misc inconsistencies (14 items)

### ISSUE THREAD

**Issue by hfrick** [tidymodels/workshops#489]
These came out of rebuilding the classwork files against the 2026 decks. Each one was verified by reading the cited line; line numbers are as of `classwork-2026` (checked 2026-08-19).

Grouped by severity: items 1-7 are broken or incorrect, 8-12 are stale leftovers from the intro/advanced split, 13-14 are minor.

Several of these already have a workaround on the classwork side (noted inline as *Classwork:*), so the classwork files are not blocked on them - but the slides still show the problem.

Not included here, tracked separately in `.claude/slide-review-findings.md`: the open decisions on header capitalization and setup-section naming, and the follow-ups from the install-list audit.

### 1. `02-data-budget.qmd:34` — factor levels typo

```
- A factor column (`class`) with levels `"class_1"` and `"class_1"`
```

The second should be `"class_2"` (confirmed by `04-evaluating-models.qmd:104`, which builds
`factor("class_1", levels = c("class_1", "class_2"))`).

### 2. `04-evaluating-models.qmd:767` — Your turn references an object that doesn't exist

> Repredict the training set using `final_fit`.

`final_fit` appears exactly once in the whole deck — on this line. No chunk ever creates it. The
object the deck actually builds at that point is `rf_res` (`:745`), and `last_fit()` moved out of
deck 04 entirely into deck 05. **This exercise cannot be run as written.**

Likely intended: fit `rf_wflow` to `cls_train` (e.g. `rf_fit <- fit(rf_wflow, cls_train)`) and
`augment()` that on the training set.

*Classwork:* `classwork-04-evaluating-models.qmd` rewords the prompt to "using the random forest
model" so the exercise is doable.

### 3. `04-evaluating-models.qmd:777` — the whole-game status update is deck 05's, and deck 04's own is stranded in Backup Slides

There are exactly four `whole-game-transparent-*.jpg` images and exactly four status-update decks, so
the intended mapping is unambiguous. Actual state:

| Deck | Line | Header | Chunk label | Image | |
|---|---|---|---|---|---|
| 02 | 224 | `- status update` | `diagram-split` | `…transparent-split` | ok |
| 03 | 732 | `- status update` | `diagram-model-1` | `…transparent-model-1` | ok |
| **04** | **777** | `The whole game` | `diagram-final-performance` | `…transparent-select` | **wrong, main deck** |
| **04** | **984** | `- status update` | `diagram-resamples` | `…transparent-resamples` | **right, but in Backup** |
| 05 | 430 | `- status update` | `diagram-select` | `…transparent-select` | ok |
| 05 | 542 | `The whole game` | `diagram-final-performance` | `whole-game-final-performance` | ok, finale |

Three things are wrong with `:777`:

1. **Wrong image.** It shows `transparent-select`, the *select* stage — that is deck 05's step, and
   deck 05 shows the identical image at `:430`. Deck 04 teaches resampling, so it should be
   `transparent-resamples`. A learner finishing deck 04 is shown progress they have not made yet,
   then sees the same diagram again in deck 05.
2. **Wrong chunk label.** `diagram-final-performance` is the label deck 05 uses at `:544` for the
   actual finale.
3. **Header** is `## The whole game`, missing the `- status update` suffix that 02, 03 and 05:430
   all carry.

All three together suggest `:777–783` was copy-pasted from deck 05's closing slide with only the
image path changed.

Deck 04's *correct* slide — right header, right label, right image — is at `:984`, inside
`# Backup Slides` (starts at `:785`), positioned after "Bootstrapping". That placement made sense
when the alternate resampling schemes were part of the main flow; they are backup material now, and
the status update was carried along with them.

**Fix:** move `:984–990` up to replace `:777–783`, and delete the leftover. That restores the
02 → 03 → 04 → 05 progression (split → model-1 → resamples → select) and leaves deck 05 owning both
`select` and the `final-performance` finale.

Nothing to change in the classwork: these are all `echo: false` image chunks, correctly excluded.

### 4. Seed mismatch for `cls_folds` between decks 04 and 05

- `04-evaluating-models.qmd:555–556` — `set.seed(343)`, then `cls_folds <- vfold_cv(cls_train, v = 10)`
- `05-tuning-models.qmd:40–41` — `set.seed(123)`, then `cls_folds <- vfold_cv(cls_train, v = 10)`

Deck 05's slide claims to reproduce deck 04's state but produces **different folds**. Any resampled
numbers a learner carries from deck 04 into deck 05 won't match. Pick one seed and use it in both.

*Classwork:* each file mirrors its own deck, so it inherits the mismatch.

### 5. `05-tuning-models.qmd:440` — text says random forest, code fits a boosted tree

> Suppose that we are happy with our random forest model.

Ten lines later (`:448`) the code is `bst_final_wflow <- finalize_workflow(bst_wflow, best_parameter)`.
Deck 05 is an xgboost story throughout; this sentence is left over from the pre-2026 version.

### 6. `05-tuning-models.qmd:287–297` — stale chunk, and the code does not run

```r
#| label: nnet-cal-plot     # <- deck 05 has no neural network
#| eval: false
bst_res |>
  cal_plot_windowed(
    # truth = class,
    # estimate = .pred_event,     # <- .pred_event is the old advanced-deck column
    window_size = 0.2,
    step_size = 0.025,
  )
```

Copied from the deleted `advanced-02` material. Two problems:

1. The commented-out column name `.pred_event` is from the old data; the 2026 data uses
   `.pred_class_1`.
2. **The code errors.** Because the chunk is `eval: false`, nobody has run it. Verified against
   probably 1.2.0 / tune 2.1.0:

   ```
   Error in `combine_vars()`:
   ! At least one layer must contain all faceting variables: `.config`
   ```

   `cal_plot_windowed()` on a multi-candidate `tune_results` fails, and passing
   `parameters = select_best(...)` does **not** help. What does work:

   ```r
   # what the postprocessing deck does (extras-postprocessing.qmd:334)
   bst_res |>
     collect_predictions(parameters = select_best(bst_res, metric = "brier_class")) |>
     cal_plot_windowed(truth = class, estimate = .pred_class_1,
                       window_size = 0.2, step_size = 0.025)
   ```

   The deck's *other* calibration chunk, `test-cal` at `:477` (`final_res |> cal_plot_windowed()`),
   is executed and works fine — `last_fit()` results have a single configuration.

*Classwork:* mirrors the deck and keeps the chunk `eval: false`, so learners see the code but don't
hit the error.

### 7. `extras-postprocessing.qmd:149–152` — the `three_rows` tibble is half-migrated

```r
three_rows <- 
  tribble(
     ~ class, ~ .pred_class, ~.pred_class_1, ~.pred_class_2,
     "event",       "event",          0.6,             0.4,     # <- values not migrated
     "event",    "nonevent",          0.4,             0.6, 
  "nonevent",    "nonevent",          0.1,             0.9  
  ) |> 
  mutate(across(where(is.character), factor))
```

The **column names** were renamed to `.pred_class_1` / `.pred_class_2`, but the **values** are still
`"event"` / `"nonevent"` from the deleted advanced deck. The deck's own data has levels `class_1` /
`class_2` (verified: `sim_classification(intercept = -10)` |> `levels(class)`), and the rest of the
deck uses `.pred_class_1` correctly (`:341`, `:470–471`).

This undercuts the teaching point. The comment on `:162` says
`.pred_class_1:.pred_class_2  # No argument name and order matches factor levels` — the whole idea is
that those columns map positionally onto the outcome's factor levels. With levels named
`event`/`nonevent` and columns named `class_1`/`class_2`, that correspondence is invisible.

It happens to work, but only by luck of alphabetical ordering: `event` < `nonevent`, so level 1 is
`event` and `.pred_class_1` really is P(level 1).

**Fix:** use `"class_1"` / `"class_2"` for the values. Verified behaviour-preserving — `predict()`
returns `event, event, nonevent` before and `class_1, class_1, class_2` after, i.e. the same
positions, so row 2 is still the row whose hard prediction gets flipped by the 1/3 threshold. That
flip is the point of the example and it survives.

Note the prose uses "event" as a *concept* at `:108`, `:112`, `:349`, `:361` ("what it means to be an
event"). That is intentional and should **not** be renamed — only the factor values in this tibble.

*Classwork:* already applied in `classwork-extras-postprocessing.qmd`.

---

### 8. `02-data-budget.qmd` — chunk labels still named after the retired dataset

`:139` `forested-split`, `:160` `forested-train-test`, `:167` `forested-train`,
`:198` `forested-split-prop`, `:215` `ex-explore-forested` (countdown `id = "explore-forested"`).

Not learner-visible, but they name a dataset the deck no longer uses.

### 9. `06-feature-engineering-part-one.qmd:51` — chunk label typo

`#| label: glimse-leaf-data` → `glimpse-leaf-data`.

### 10. `06-feature-engineering-part-one.qmd:1053` and `:1107` — lowercase method names in headers

```
## umap with recipes      <- should be "UMAP"
## isomap with recipes    <- should be "Isomap"
```

Every other mention in the deck is capitalized correctly — `## UMAP` (`:935`), `## UMAP Algorithm`
(`:941`), `## UMAP parameters` (`:953`), `## UMAP hesitancy` (`:995`), `## Isomap` (`:1066`),
`## Isomap Algorithm` (`:1076`), `## Isomap Pros and Cons` (`:1085`), and the bullets at `:614–615`.
Only the two "with recipes" headers are lowercase, so this is a slip rather than a choice.

Canonical forms, confirmed from the packages rather than from usage:

- `embed::step_umap()` sets `prefix = "UMAP"`; docs say "the number of UMAP components". UMAP is an
  acronym (Uniform Manifold Approximation and Projection), so it is fully capitalized.
- `recipes::step_isomap()` sets `prefix = "Isomap"`; docs say "Isomap is a form of multidimensional
  scaling (MDS)" and it generates columns `Isomap1`, `Isomap2`, …

So it is **Isomap**, not `ISOmap` or `ISOMAP` — the name contracts "isometric mapping", it is not an
acronym. The column names learners see in the output of these very chunks are `UMAP1`/`UMAP2` and
`Isomap1`/`Isomap2`, a good reason for the prose to match.

*Classwork:* already uses `## UMAP with recipes` and `## Isomap with recipes`.

### 11. `06-feature-engineering-part-one.qmd:39` — data loaded from a 2025 branch URL

```r
"https://github.com/tidymodels/workshops/raw/refs/heads/2025-GMOFETML/slides/leaf_data.RData"
```

`slides/leaf_data.RData` is tracked on `main` today (`git ls-files` confirms). The deck fetches it
over the network from a stale workshop branch instead of using the local copy. If that branch is ever
deleted, deck 06 breaks for everyone. The classwork file copies the same URL, so both would break.

### 12. `07-feature-engineering-part-two.qmd:604` — missing word in a Your turn

> *The lubridate package might provide helpful*

Probably "might provide helpful **functions**" / "might be helpful".

*Classwork:* reads "might be helpful".

---

### 13. `04-evaluating-models.qmd` — `probably` is only loaded in a hidden chunk

The only executed `library(probably)` is at `:171`, inside an `echo: false` chunk. It *is* shown to
learners at `:236`, but only inside a non-executed ```` ```r ```` illustration block in a Your turn.
The deck's own runnable `cal_plot_windowed()` at `:755` therefore depends on a load the learner never
sees. Worth adding `library(probably)` to a visible chunk.

*Classwork:* adds it to `## Setup` explicitly.

### 14. `04-evaluating-models.qmd:103–104` — `%>%` where the rest of the deck uses `|>`

Cosmetic, and defensible: `01-introduction.qmd:77` explicitly tells learners "You can use the
magrittr `%>%` or base R `|>` pipe". The only other `%>%` uses in the deck are at `:301` and `:348`,
both in hidden chunks. Flagging for consistency, not correctness.
